File: RPi Experiments/RPi2-1/Python/04_slideSwitchLed.py
import RPi.GPIO as GPIO
import time

ledPin = 11 # physical pin number for the LED
slidePin = 12 # physical pin number for the button

def setup():
    GPIO.setwarnings(False) # don't show warnings if GPIO is in use
    GPIO.setmode(GPIO.BOARD) # use GPIO by physical location
    GPIO.setup(ledPin,GPIO.OUT) # initialize the LED pin as output
    GPIO.setup(slidePin,GPIO.IN, pull_up_down=GPIO.PUD_UP) # initialize the button pin as input and use internal resistor
    GPIO.output(ledPin,GPIO.HIGH) # turn off led
    
def handleLed(ev=None):
    if GPIO.input(slidePin):
        GPIO.output(ledPin,GPIO.HIGH) # turn off led
    else:
        GPIO.output(ledPin,GPIO.LOW) # turn on led

def loop():
	while True:
		pass
        
def destroy():
	GPIO.output(ledPin,GPIO.HIGH) # turn off led
	GPIO.cleanup() # release resource

if __name__ == '__main__': # program start from here
    try:
        setup()
        loop()

    except KeyboardInterrupt: # when 'Ctrl+C' is pressed, the child program destroy() will be  executed.
        destroy()

File: RPi Experiments/RPi2-1/C/04_slideSwitchLed.c
#include<wiringPi.h>
#include<stdio.h>

#define ledPin 0 //wiringPi pin number for the LED
#define slidePin 1 //wiringPi pin number for the button

int main(void)
{
	if(wiringPiSetup() == -1)
	{
		printf("Setup wiring Pi failed!");
		return 1;
	}
	
	pinMode(ledPin,OUTPUT); //initialize the LED pin as output
	pinMode(slidePin,INPUT); //initialize the button pin as input
	
	while(1)
	{
		digitalWrite(ledPin, HIGH); //turn off the LED
		if(digitalRead(slidePin) == 0) //read the button state, 0 if pressed
		{
            digitalWrite(ledPin, LOW); //turn on the LED
		}
	}
	
	return 0;
}